core/src/apps/solana/transaction/parse.py

```diff
@@ -0,0 +1,227 @@
+from typing import TYPE_CHECKING
+
+from trezor.crypto import base58
+from trezor.wire import ProcessError
+
+from apps.common.readers import read_compact_size, read_uint32_le, read_uint64_le
+
+from ..constants import (
+    ADDRESS_READ_ONLY,
+    ADDRESS_RW,
+    ADDRESS_SIG,
+    ADDRESS_SIG_READ_ONLY,
+    ADDRESS_SIZE,
+)
+
+if TYPE_CHECKING:
+    from typing import Callable
+
+    from trezor.utils import BufferReader
+
+    from ..types import Address, AddressReference, InstructionIdFormat, RawInstruction
+
+
+def assert_cond(condition: bool, message: str = "") -> None:
+    if not condition:
+        raise ProcessError(message)
+
+
+def parse_header(serialized_tx: BufferReader) -> tuple[bool, int, int, int, int]:
```

Comment:
Do we really want to move this out from parse.py? Our intention was to keep all parsing function in one file.

Reply:
thing is, `Transaction.__init__` is arguably a "parse function". so to me it would make much more sense to move this and friends ("parse addresses" "parse instructions" "parse lookup tables") and only keep things like "parse pubkey", "parse enum", that actually decode a certain data type

Reply:
[Changes](https://github.com/trezor/trezor-firmware/commit/bf34dd0275f7838ce35b3448d987fbf916314553)